sdcm/cluster.py

```diff
@@ -707,6 +707,7 @@ def wait_ssh_up(self, verbose=True, timeout=500):
         if verbose:
             text = '%s: Waiting for SSH to be up' % self
         wait.wait_for(func=self.remoter.is_up, step=10, text=text, timeout=timeout, throw_exc=True)
+        assert self.remoter.is_up(), "Network still isn't up after waiting"
```

Comment:
this code will run only when the line above(710) didn't throw an exception, that means that is_up succeed so why to check is_up again? and if wait failed it will throw an exception and we never get to the proposed assert.
@amoskong what did you want to achieve by this patch?

Reply:
Exception would be suppressed in is_up(), it won't raise exception even network isn't up.

```
    def is_up(self):
        """
        Check if the remote host is up.

        :return: True if the remote host is up, False otherwise
        """
        try:
            self._ssh_ping()
        except process.CmdError as e:
            self.log.error("Error while executing SSH command: %s" % e.result)
            return False
        except Exception, details:
            self.log.error('Error checking if SSH is up: %s', details)
            return False
        else:
            return True
```

Reply:
`wait_for` raises `WaitTimeoutExpired` when `throw_exc=True` and waiting time is up

Reply:
I saw an exception raised from install_sct_log_formatter which is after wait_for(self.remoter.is_up....)

You are right, I just found another enterprise log, I saw a timeout exception raised from wait_for().

The following exception means the network really up in wait_for(), but it's unavailable in install sct log formatter file.

```
2019-01-13 21:04:09,634 output           L0642 DEBUG| Exception in thread Thread-82243:
2019-01-13 21:04:09,635 output           L0642 DEBUG| Traceback (most recent call last):
2019-01-13 21:04:09,635 output           L0642 DEBUG|   File "/usr/lib64/python2.7/threading.py", line 812, in __bootstrap_inner
2019-01-13 21:04:09,635 output           L0642 DEBUG|     self.run()
2019-01-13 21:04:09,635 output           L0642 DEBUG|   File "/usr/lib64/python2.7/threading.py", line 765, in run
2019-01-13 21:04:09,635 output           L0642 DEBUG|     self.__target(*self.__args, **self.__kwargs)
2019-01-13 21:04:09,636 output           L0642 DEBUG|   File "/sct/sdcm/cluster.py", line 651, in backtrace_thread
2019-01-13 21:04:09,636 output           L0642 DEBUG|   File "/sct/sdcm/cluster.py", line 632, in get_backtraces
2019-01-13 21:04:09,636 output           L0642 DEBUG|   File "/sct/sdcm/cluster.py", line 699, in wait_ssh_up
2019-01-13 21:04:09,636 output           L0642 DEBUG|   File "/sct/sdcm/cluster.py", line 464, in install_sct_log_formatter
2019-01-13 21:04:09,636 output           L0642 DEBUG|   File "/sct/sdcm/remote.py", line 686, in send_files
2019-01-13 21:04:09,637 output           L0642 DEBUG|   File "/sct/sdcm/remote.py", line 805, in ssh_run
2019-01-13 21:04:09,637 output           L0642 DEBUG| CmdError: Command 'scp -r  -o StrictHostKeyChecking=no -o BatchMode=yes -o ConnectTimeo
 -o UserKnownHostsFile=/var/tmp/tmp6wtBVj -P 22 -i /root/.ssh/scylla-qa-ec2 /sct/data_dir/sct_log_formatter 'centos@172.16.99.75:"/var/tmp"''
```

Reply:
@roydahan FYI.  it means in 1tb longevity, one aws instance is created, and network is up in first try, when we try to copy files to it the network is down for some unknown reason.

Reply:
I know, I couldn't understand why it happened.
We caught this too late and the node was already removed due to another terminate nemesis that ran on it.

I suspect it was an issue with AWS but I can't tell, we don't have enough info.
If it happens again, we need to connect the node and check it's logs.